File: 백준/Gold/5485. 평균값 수열/평균값 수열.cc
#include<bits/stdc++.h>
using namespace std;
using ll = long long;

int main(){
    cin.tie(0); cout.tie(0); ios::sync_with_stdio(false);
    int n; cin >> n;
    ll low, high, cur; cin >> low >> high; cur = high;
    for(int i = 2; i < n; i++){
        ll nxt, tmp = low; cin >> nxt;
        low = 2*cur-high;
        high = min(2*cur-tmp, nxt);
        cur = nxt;
    }
    cout << max(0ll, high-low+1);
    return 0;
}

File: 백준/Gold/5485. 평균값 수열/평균값 수열.py
import sys
def print(*args, sep=" ", end="\n"): return sys.stdout.write(sep.join(map(str, args)) + end)
def input(): return sys.stdin.readline().rstrip()

def main():
    n = int(input())
    low = int(input())
    high = cur = int(input())
    for _ in range(2, n):
        nxt = int(input())
        low, high = 2*cur-high, min(2*cur-low, nxt)
        cur = nxt
    print(max(0, high-low+1))
    return 0
if __name__ == '__main__':
    sys.exit(main())